OPM/opm-rule-onboarding.ps1

```diff
@@ -0,0 +1,540 @@
+# - No validation checking on PlatformID
+# - No validation checking on existing rules
+# - No validation checking on command restrictions
+# - Capability to add OPM rules only, no update OPM rule capability
+#
+###########################################################################
+[CmdletBinding(DefaultParametersetName="Create")]
+param
+(
+	[Parameter(Mandatory=$true,HelpMessage="Please enter your PVWA address (For example: https://pvwa.mydomain.com/PasswordVault)")]
+	#[ValidateScript({Invoke-WebRequest -UseBasicParsing -DisableKeepAlive -Uri $_ -Method 'Head' -ErrorAction 'stop' -TimeoutSec 30})]
+	[Alias("url")]
+	[String]$PVWAURL,
+	
+	[Parameter(Mandatory=$false,HelpMessage="Enter the Authentication type (Default:CyberArk)")]
+	[ValidateSet("cyberark","ldap","radius")]
+	[String]$AuthType="cyberark",
+	
+	[Parameter(Mandatory=$false,HelpMessage="Enter the RADIUS OTP")]
+	[ValidateScript({$AuthType -eq "radius"})]
+	[String]$OTP,
+	
+	[Parameter(Mandatory=$false,ValueFromPipeline=$false,ValueFromPipelineByPropertyName=$true)]
+	[ValidateScript({Test-Path $_})]
+	[Alias("path")]
+	[String]$CsvPath,
+	
+	[Parameter(Mandatory=$false,ValueFromPipeline=$false,ValueFromPipelineByPropertyName=$true)]
+	[ValidateSet("Comma","Tab")]
+	[Alias("delim")]
+	[String]$CsvDelimiter = "Comma",
+	
+	# Use this switch to Disable SSL verification (NOT RECOMMENDED)
+	[Parameter(Mandatory=$false)]
+	[Switch]$DisableSSLVerify
+)
+
+# Get Script Location 
+$ScriptLocation = Split-Path -Parent $MyInvocation.MyCommand.Path
+
+# Set Log file path
+$LOG_FILE_PATH = "$ScriptLocation\OPM_Rule_Onboarding_Utility.log"
+
+$InDebug = $PSBoundParameters.Debug.IsPresent
+$InVerbose = $PSBoundParameters.Verbose.IsPresent
+
+# Global URLS
+# -----------
+$URL_PVWABaseAPI = $PVWAURL+"/WebServices/PIMServices.svc"
+$URL_PVWAAPI = $PVWAURL+"/api"
+$URL_Authentication = $URL_PVWAAPI+"/auth"
+$URL_Logon = $URL_Authentication+"/$AuthType/Logon"
+$URL_Logoff = $URL_Authentication+"/Logoff"
+
+# URL Methods
+# -----------
+$URL_PlatformCommands = $URL_PVWABaseAPI+"/Policy/{0}/PrivilegedCommands"
+
+
+# Script Defaults
+# ---------------
+$g_CsvDefaultPath = $Env:CSIDL_DEFAULT_DOWNLOADS
+
+
+# Initialize Script Variables
+# ---------------------------
+$g_LogonHeader = ""
+
+#region Functions
+Function Test-CommandExists
+{
+    Param ($command)
+    $oldPreference = $ErrorActionPreference
+    $ErrorActionPreference = 'stop'
+    try {if(Get-Command $command){RETURN $true}}
+    Catch {Write-Host "$command does not exist"; RETURN $false}
+    Finally {$ErrorActionPreference=$oldPreference}
+} #end function test-CommandExists
+
+Function Encode-URL($sText)
+{
+	if ($sText.Trim() -ne "")
+	{
+		Log-Msg -Type Debug -Msg "Returning URL Encode of $sText"
+		return [System.Web.HttpUtility]::UrlEncode($sText)
+	}
+	else
+	{
+		return $sText
+	}
+}
+
+Function Get-TrimmedString($sText)
+{
+	if($sText -ne $null)
+	{
+		return $sText.Trim()
+	}
+	# Else
+	return $sText
```

Comment:
Not sure where you are using this or what you expect, but think if it is OK to return NULL
Maybe if the text is null, return "N/A" instead

Reply:
For the purposes of the API input, null would be better than N/A. This part of the code was unchanged from your code and how you were using it in the Account Onboarding script.